03/app.js

```diff
@@ -11,4 +11,28 @@ const buttonSettings = {
         color: '#444'
     },
     text: 'Click me!',
-}
\ No newline at end of file
+}
+
+
+const button = document.createElement('button')
+const parent = document.querySelector(".parent-for-button")
+
+for (const key in buttonSettings) {
+
+    const properties = buttonSettings[key]
+
+    if (key === "attr") {
+        for (const item in properties) {
+            button.setAttribute(item, properties[item])
+        }
```

Comment:
Skoro i tak mamy `if` więc nie wiele zyskujemy na pętli zewnętrznej - to tylko zaciemnia kod.
Zrezygnowałbym z niej, a wykorzystał po prostu właściwości tj. `for (const item in buttonSettings.attr) {`